### PR TITLE
GIV-92 Upgraded `RunnableTask` to work with Play 2.4.x

### PR DESCRIPTION
**Pull request by ericluria**

### CODE REVIEW

**Comment by ericluria** [comment]
To test locally,

1. Change the version in `build.sbt` to `1.0.4-LOCAL`
2. Open another project (using Play 2.4.x) and include it as a dependency: `"com.jaroop" %% "play-sbt-tasks" % "1.0.4-LOCAL"`
3. `sbt console`
4. `import com.jaroop.play.sbt._`
5. 
```
class TestTask(arguments: String) extends RunnableTask(arguments) {
    def execute(): Unit = println(s"Arguments: $arguments")
}
```
6. `val test = new TestTask("testing 1 2 3")`
7. `test.run()`
8. Voila!

File: build.sbt [hunk 1]
```diff
@@ -34,21 +36,10 @@ lazy val commonSettings = Seq(
...
-    .settings(commonSettings: _*)
-    .settings(
-        name := "play-sbt-tasks",
-        crossScalaVersions := Seq("2.11.6", "2.10.5"),
-        libraryDependencies ++= Seq(
-          "com.typesafe.play" %% "play" % "2.3.8"
-        )
-    )
-
-lazy val plugin = (project in file("plugin"))
```
Comment (mhzajac) [inline build.sbt]: Why was `plugin` deleted?

File: src/main/scala/RunnableTask.scala [hunk 2]
```diff
@@ -11,10 +10,13 @@ abstract class RunnableTask(arguments: String) extends Runnable {
...
-    /** Run the task within a static Play application, and shut it down when finished. */
+    /** Run the task within an ad-hoc Play application, and shut it down when finished. */
     def run(): Unit = {
-        val application: StaticApplication = new StaticApplication(new java.io.File("."))
-        try { this.execute() } finally { Play.stop() }
+		val env = Environment(new java.io.File("."), this.getClass.getClassLoader, Mode.Dev)
+		val context = ApplicationLoader.createContext(env)
+		val loader = ApplicationLoader(context)
+		val app = loader.load(context)
+        try { this.execute() } finally { Play.stop(app) }
```
Comment (mhzajac) [inline src/main/scala/RunnableTask.scala]: Indentation is messed up

File: build.sbt [hunk 3]
```diff
@@ -34,21 +36,19 @@ lazy val commonSettings = Seq(
...
-    .settings(
-        name := "play-sbt-tasks-plugin",
-        scalaVersion := "2.10.5",
-        sbtPlugin := true
-    )
+  .settings(sonatypeSettings: _*)
+  .settings(commonSettings: _*)
+  .settings(
+    name := "play-sbt-tasks-plugin",
+    scalaVersion := "2.11.7",
```
Comment (mhzajac) [inline build.sbt]: sbt plugins can only use 2.10

File: build.sbt [hunk 4]
```diff
@@ -1,8 +1,10 @@
 import SonatypeKeys._
 
+scalaVersion := "2.11.7"
```
Comment (mhzajac) [inline build.sbt]: Shouldn't this go in `root`?

Comment (ericluria) [inline build.sbt]: Yeah, you're right. Fixed.

**Pull request merged**